homeassistant/components/rituals_perfume_genie/config_flow.py

```diff
@@ -0,0 +1,60 @@
+"""Config flow for Rituals Perfume Genie integration."""
+import logging
+
+from aiohttp import ClientResponseError
+from pyrituals import Account, AuthenticationException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import ACCOUNT_HASH, DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Rituals Perfume Genie."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA)
+
+        errors = {}
+
+        session = async_get_clientsession(self.hass)
+        account = Account(user_input[CONF_EMAIL], user_input[CONF_PASSWORD], session)
+
+        try:
+            await account.authenticate()
+        except ClientResponseError:
+            errors["base"] = "cannot_connect"
+        except AuthenticationException:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            await self.async_set_unique_id(account.data[CONF_EMAIL])
```

Comment:
The unique ID is case sensitive... email, however, is not.

Reply:
I noticed this and then fixed it in commit: https://github.com/home-assistant/core/pull/46218/commits/2618110376772a581771c2cf5e1226fb268e72bc Fix config_entry unique_id when using uppercase letters.
We do not use the entered email `user_input[CONF_EMAIL]` but the email returned by the API `account.data[CONF_EMAIL]`.
The case will always be the same unless the email has been changed on the rituals servers.